tests/opentelemetry-test-utils/src/opentelemetry/test/asgitestutil.py

```diff
@@ -74,3 +75,40 @@ def get_all_output(self):
             except asyncio.TimeoutError:
                 break
         return outputs
+
+
+class AsyncAsgiTestBase(TestBase, IsolatedAsyncioTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.scope = {}
+        setup_testing_defaults(self.scope)
+        self.communicator = None
+
+    def tearDown(self):
+        if self.communicator:
+            asyncio.get_event_loop().run_until_complete(
+                self.communicator.wait()
+            )
+
+    def seed_app(self, app):
+        self.communicator = ApplicationCommunicator(app, self.scope)
+
+    async def send_input(self, message):
+        await self.communicator.send_input(message)
+
+    async def send_default_request(self):
+        await self.send_input({"type": "http.request", "body": b""})
+
+    async def get_output(self):
+        output = await self.communicator.receive_output(0)
```

Comment:
Using something bigger than 0 helps in getting some more tests green

Reply:
Updated to support a custom timeout value so we can control in the test instead of touching the code here. In tests I'm assuming timeout 0.01